src/pymmcore_plus/core/_mmcore_plus.py

```diff
@@ -1734,24 +1736,32 @@ def saveSystemConfiguration(self, filename: str) -> None:
         **Why Override?** To also save pixel size configurations.
         """
         super().saveSystemConfiguration(filename)
-        px_configs = self.getAvailablePixelSizeConfigs()
-        if not px_configs:
-            return
         # saveSystemConfiguration does not save the pixel size config so here
         # we add to the saved file also any pixel size config.
         with open(filename, "a") as f:
-            f.write("# PixelSize settings")
-            for px_config in px_configs:
-                data = self.getPixelSizeConfigData(px_config)
-                obj = data.dict()["Objective"]["Label"]
-                px_size = self.getPixelSizeUmByID(px_config)
-                px_affine = self.getPixelSizeAffineByID(px_config)
-                cfg = (
-                    f"\nConfigPixelSize,{px_config},Objective,Label,{obj}\n"
-                    f"PixelSize_um,{px_config},{px_size}\n"
-                    f"PixelSizeAffine,{px_config},{','.join(map(str, px_affine))}"
+            self._save_pixel_configurations(f)

```

Comment:
this will open and close the file unnecessarily if self.getAvailablePixelSizeConfigs() doesn't return anything.  Also, though it's a private method, technically `TextIOBase` doesn't "demand" that the file have been opened in "append" mode, so the decoupling there leads to the (admittedly tiny) possibility that `_save_pixel_configuration` would overwrite the full context of the file (and not just append).  So, change `_save_pixel_configuration` to take a filename instead of an open handle, and move the open() call inside of that method